packages/core/src/dialects/abstract/query-interface.d.ts

```diff
@@ -299,6 +302,13 @@ export type CreateTableAttributes<
   [name in keyof TAttributes]: DataType | CreateTableAttributeOptions<M>;
 };
 
+/**
+ * Constraint on a table
+ */
+export type Constraint = {
```

Comment:
prefer interfaces

Reply:
Happy to change it, though I'm curious why? I thought interfaces were supposed to represent "doing" something

Reply:
I think we decided on just using interfaces as much as possible so it's a consistency thing more than performance wise.

Reply:
Pretty much yes. The entire project uses interfaces when possible, with https://github.com/sequelize/sequelize/blob/main/packages/core/src/dialects/abstract/query-generator.d.ts as the only exception because I did not know at the time and we never took the time to fix that mistake